ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/CachingDamlLedgerStateReader.scala

```diff
@@ -37,9 +36,12 @@ class CachingDamlLedgerStateReader(
 
   override def readState(keys: Seq[DamlStateKey]): Future[Seq[Option[DamlStateValue]]] = {
     this.synchronized { readSet ++= keys }
-    val cachedValues = cache.getAllPresent(keys.asJava).asScala.map {
-      case (key, value) => key -> Some(value)
-    }
+    val cachedValues = keys
+      .map { key =>
+        key -> cache.getIfPresent(key)
+      }
```

Comment:
A chain like this would probably benefit from a `.view`.

```suggestion
      .view
      .map(key => key -> cache.getIfPresent(key))
```

Reply:
Good point -- changed accordingly.